testsuite/tests/singlecluster/gateway/reconciliation/listeners/test_gateway_listeners_dns.py

```diff
@@ -0,0 +1,42 @@
+"""Testing specific bug that happens when listener hostname in Gateway gets changed while DNSPolicy is applied."""
+
+import pytest
+
+from testsuite.gateway import GatewayListenerTls
+from testsuite.utils import is_nxdomain, sleep_ttl
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy, pytest.mark.tlspolicy]
+
+
+@pytest.fixture(scope="module")
+def domain_prefix():
+    """Prefix for this test"""
+    return "dnsbug"
+
+
+@pytest.mark.issue("https://github.com/Kuadrant/kuadrant-operator/issues/794")
+def test_change_listener(custom_client, check_ok_https, gateway, route, second_domain, wildcard_domain):
+    """
+    This test checks if after change of listener hostname in a Gateway while having DNSPolicy applied, that
+    the old hostname gets deleted from DNS provider. After editing the hostname in HTTPRoute to the new value
+    this test checks the reconciliation of such procedure.
+
+    WARNING
+        Running this test in unpatched Kuadrant will leave orphaned DNS records in DNS provider if the test fails
+        on the last assert. If you want to delete the records you need to do it manually.
+        The orphaned DNS records will contain string 'dnsbug'
```

Comment:
Its true the that the bug was fixed but I want to keep this warning in the test if the bug would ever be reintroduced to inform the tester of the orphaned dns records.

Reply:
I think that it does not make sense to have such warnings in general because a lot of bugs can potentially be (re)introduced and our code base would be full of such warnings quickly. However, I don't mind keeping this one there, I would maybe rephrase a little. It is easy to understand this as if the bug is still not fixed which is not true. Up to you, I approve this even with current wording.

Reply:
After a bit of consideration I agree with you. There are many tests in testsuite that can leave leftovers in cluster, SSO provider or now DNS provider. I think it should be rather expected so this warning is a bit out of place. Will remove it.